freqtrade/freqtradebot.py

```diff
@@ -394,21 +394,21 @@ def create_trade(self, pair: str) -> bool:
             logger.info(f"Pair {pair} is currently locked.")
             return False
 
+        if not self.get_free_open_trades():
+            logger.debug(f"Can't open a new trade for {pair}: max number of trades is reached.")
+            return False
```

Comment:
I guess this code is ok here - no need to check for buy-signals when no trade-slot is open.
however, please add a clarifying comment here (i've already had a comment written that this segment is not needed and can never be hit) if you don't mind.

It's called in line #154, and if it returns false, `enter_positions()` and `create_trade()` are never called.

However, if the first pair in the whitelist opens a trade, then this is necessary for subsequent pairs - to avoid buying 5 trades in one iteration while max_open_trades is set to 3.

Now my wording is a bit ... verbose - so if you're able to find something concise it's even better (probably something along the lines of `# Prevent opening too many trades within one iteration` ...

Reply:
Gotcha, comment added, and adjustments made to this PR. Sorry, this took a while, haven't had much time recently.

Reply:
Gotcha, comment added, and adjustments made to this PR. Sorry, this took a while, haven't had much time recently.